bitrise.yml

```diff
@@ -38,6 +40,7 @@ workflows:
         - file_type: ".swift"
     - script@1.1:
         title: Build for Testing
+        run_if: '{{getenv "RUN_UI_TESTS" | eq "Run_UI_Tests"}}'
```

Comment:
I would remove this from here, because I think we should always build the not matter the file we change. And so, this line would go to the step below: `xcode-test-without-building@0` so that we don't run the unit tests, and the rest of UI tests as you have already configured

Reply:
@jjSDET I suggested removing this line from here BUT adding it in to conditionally run the Unit Tests in this line: https://github.com/mozilla-mobile/firefox-ios/pull/14088/files#diff-61d1ee831cb1df242161a7d23171914a1d5d628c7659b52db957d80a9394bd68L48

Please add it there.

Reply:
removed it from `Build for Testing` and added it to `xcode-test-without-building`

Reply:
We may need to add it to other steps that depends on the test-without-building step, for example, if there is no output here, the danger step fails... sorry I just realized, I can push that change to the PR

Reply:
And rebase with latest main so that we pick the changes to run the tests on M1 as that's probably the cause of those failures that you are seeing